### ISSUE TITLE
"Stranger Things" doesn't appear on any list

### ISSUE THREAD

**Issue by mp-pinheiro** [asciidisco/plugin.video.netflix#222]
*I'm submitting a ...*
  - [X] bug report
  - [ ] feature request
  - [ ] support request

## General infomration
The show "Stranger Things" does not appear on any lists whatsoever.

### Prerequisites
None.

* [X] Are you running the latest version?

### Description

Stranger Things appears to me on my PC on the lists Netflix Originals, Top Picks, New Releases and on My List, but doesn't appear on any of those in the Kodi plugin.

I added it on My List (which only has 1 show, Friends), and only Friends appears.

I also tried the Search feature, does not return the show (tried searching for "stranger things" and "Stranger Things").

### Steps to Reproduce

1. Open any of the lists described above
2. Look for Stranger Things

**Expected behavior:** Find and watch Stranger Things.

**Actual behavior:** Can't find the show.

### Installation

* [ ] I installed the plugin via zip from the Releases page
* [X] I´m using the Netflix Repo
* [ ] I´m using a different source (Please tell which)

_I intalled directly from source code the first time, did not work as well. So I decided to unninstal and install from repo to receive updates. I deleted all files before re-installing._

### Operating System

* [ ] Linux (x86/x64)
* [ ] OSX (x86/x64)
* [ ] Windows (x86/x64)
* [X ] Linux (ARM Pi2) 
* [ ] Android

#### Additional informatin on the environment

Using libreelec test build 1027: [link](https://forum.kodi.tv/showthread.php?tid=298461&pid=2660293#pid2660293)

## Debug log
Log file loggin opening the addon and selecting "My List", which shows only "Friends" and should show "Stranger Things" ass well.
[](http://paste.ubuntu.com/25839536/)

## Other information

Stranger Things season 2 was released yesterday, this could be an issue related to any new show (but I'm just speculating).

### Screenshots

[My List on PC (Chrome)](https://i.imgur.com/DXZYCCn.png)
[My List on the Pi](https://imgur.com/a/eqqRh) (Sorry for the photo)

**Comment by pinoelefante** [comment]
it's a strange(r) thing.
it happens to me too, but it appears when i use search

**Comment by pinoelefante** [comment]
seeing your log, i noticed this. maybe is related
"[Netflix] Fetched item from cache: "

**Comment by mp-pinheiro** [comment]
@pinoelefante Obligatory pun aside, you mean search like this: [https://imgur.com/a/IzODU](https://imgur.com/a/IzODU)? 

Because, as you can see, just typing "s" makes "Friends" vanish and no sign of "Stranger Things".

On the log, it says "fetched item" (1), so I added Rick and Morty to my list and it still shows only Friends.
I think it's showing only cached shows.

I'll do a wipe on the plugin and try again.

**Comment by pinoelefante** [comment]
@mp-pinheiro it's really strange, because it should appear using "Search"

**Comment by mp-pinheiro** [comment]
@pinoelefante  Alright, so I did a clean install and it did not fix it.

One thing that I thought was weird is that even after I clean installed, the addon would open with my profile pre-selected (like it cached the state of the menus). So I rebooted my Pi em voila, all 3 shows appeared: [https://imgur.com/a/LQkha](https://imgur.com/a/LQkha)

If I understand correctly, shows that you add after the menus are cached will not appear, which can be a little annoying.

I think the issue should be forked into two: one regarding the cached menus and another regarding the show not appearing on the other lists, like "Netflix Originals" and "Recently Added". What do you think?

**Comment by pinoelefante** [comment]
@mp-pinheiro i'm happy you solved your issue 😃

**Comment by asciidisco** [comment]
I´ve never considered that people could´ve been running their system without a restart for a longer period.

As I don't want to give up the caching completely for those lists, I´m going to propose a few things I´d like to do to prevent this:

- A setting to turn off the cache completely
- A lifetime for cache items (I´d say 30 minutes for lists is sufficient)
- A possibility to refresh the list you´re on (purge cache & reload)

I´d not fork this into 2 issues, as there is one reason for the behaviour.

**Comment by hawkeyexp** [comment]
I recommend a footer entry like "next page" for "refresh list"

**Comment by mp-pinheiro** [comment]
@asciidisco Sounds fantastic!

Yeah, my system is running most of the time without reboots.

So, if I understand correctly, the show is not appearing on "New Releases" and other lists as well because of the same caching "issue"?

@hawkeyexp I agree, it'd be a nice thing to have if you don't disable caching.

**Comment by asciidisco** [comment]
Do not like to have another non content list item. Would more likely use context menu items for that matter.

That reminds me of the fact that I need to write down my thoughts on better paging in Kodi (that is possible with today’s APIs) as well...

**Comment by hawkeyexp** [comment]
ok it is an alternate as context menu entry but possible some users didn't notice the option to refresh cache.

**Comment by asciidisco** [comment]
Agreed. But it wouldn’t be that bad, because the cache lifetime expiration would reveal it sooner or later.

Which also brings up the thing again about writing proper usage docs... ... ...so many todos still open.

**Comment by hawkeyexp** [comment]
ok i agree. Ps: come on on skype :-)
